src/storage/mapping/FixedContentTypeMapper.ts

```diff
@@ -21,4 +45,35 @@ export class FixedContentTypeMapper extends BaseFileIdentifierMapper {
   protected async getContentTypeFromPath(): Promise<string> {
     return this.contentType;
   }
+
+  public async mapUrlToDocumentPath(identifier: ResourceIdentifier, filePath: string, contentType?: string):
+  Promise<ResourceLink> {
+    // Handle URL suffix
+    if (this.urlSuffix) {
+      if (filePath.endsWith(this.urlSuffix)) {
+        filePath = filePath.slice(0, filePath.length - this.urlSuffix.length);
```

Comment:
slice supports negative indexes (indices? 🤔 ) so you can do `filePath.slice(0, this.urlSuffix.length)`